src/visGeometry/index.ts

```diff
@@ -1066,198 +1066,37 @@ class VisGeometry {
-     * @property this.idColorMapping uses @param colorId. It maps agent id to colorId
-     *
-     * No other module should know about this.colorsData, or the fact that it's 4 times as
-     * long as the input colors. They should only know about colorId
-     * Therefore all the public color methods should use colorId, and the private methods
-     * can convert between colorId and dataColorIndex
-     */
-
-    /**
-     * Agent color handling: private methods
-     */
-
-    private get numberOfColors(): number {
-        return this.colorsData.length / 4;
-    }
-
     private setAgentColors(): void {
         this.visAgents.forEach((agent) => {
             agent.setColor(
-                this.getColorForTypeId(agent.agentData.type),
-                this.getColorIdForTypeId(agent.agentData.type)
+                this.colorHandler.getColorInfoForAgentType(agent.agentData.type)
             );
         });
     }
 
-    private setColorArray(colors: (number | string)[]): void {
-        const colorNumbers = colors.map(convertColorStringToNumber);
-        const numberOfColors = colors.length;
-        // fill buffer of colors:
-        this.colorsData = new Float32Array(numberOfColors * 4);
-        for (let i = 0; i < numberOfColors; i += 1) {
-            // each color is currently a hex value:
-            this.colorsData[i * 4 + 0] =
-                ((colorNumbers[i] & 0x00ff0000) >> 16) / 255.0;
-            this.colorsData[i * 4 + 1] =
-                ((colorNumbers[i] & 0x0000ff00) >> 8) / 255.0;
-            this.colorsData[i * 4 + 2] =
-                ((colorNumbers[i] & 0x000000ff) >> 0) / 255.0;
-            this.colorsData[i * 4 + 3] = 1.0;
-        }
-    }
-
-    private convertDataColorIndexToId(dataColorIndex: number): number {
-        if (dataColorIndex % 4 !== 0) {
-            this.logger.error(
-                "convertDataColorIndexToId: color index not divisible by 4"
-            );
-            return -1;
-        }
-        const index = dataColorIndex / 4;
-        // this loops the index back to the beginning of the array
-        // in the chance that the index is out of range, which
-        // should be impossible. But just being cautious
-        return index % this.numberOfColors;
-    }
-
-    private getColorDataIndex(color: number[]): number {
-        /**
-         * returns the index into the colorsData array
-         */
-        const colorArray = this.colorsData;
-        const colorToCheck = map(color, (num) => round(num, 6));
-        for (let i = 0; i < colorArray.length - 3; i += 4) {
-            const index = i;
-            const currentColor = [
-                round(this.colorsData[i], 6),
-                round(this.colorsData[i + 1], 6),
-                round(this.colorsData[i + 2], 6),
-                this.colorsData[i + 3],
-            ];
-            if (isEqual(currentColor, colorToCheck)) {
-                return index;
+    public setGeom(ids: number[]): void {
```

Comment:
should this be private?

Reply:
I think this needs to be clearer as to why anyone would ever call it.  Geometry should have been initialized in a better way elsewhere.

Reply:
yeah this was previously part of : 
```  private setColorForId(id: number, colorId: number): void {
         /**
          * @param id agent id
          * @param colorId index into the numberOfColors
          */
         this.idColorMapping.set(id, colorId);

         // if we don't have a mesh for this, add a sphere instance to mesh registry?
         if (!this.visGeomMap.has(id)) {
             this.visGeomMap.set(id, DEFAULT_MESH_NAME);
         }
     }
```
I factored out the the part that sets the colorMapping, but I wasn't sure about this other part. You're right it should be private, but it also doesn't seem necessary, (we default things to spheres in the geometryStore). It might have gotten left behind accidentally during a refactor

Reply:
I see, yeah- even better if we can get away with removing it.

Reply:
I removed it